worlds/_bizhawk/README.md

```diff
@@ -0,0 +1,279 @@
+Emerald's save data is at a dynamic location in memory. It moves around when you load a new map. There is a static
+variable that holds the address of the save data, so we want to read the static variable to get the save address, and
+then use that address in a `write` to send the player an item. But between the `read` that tells us the address of the
+save data and the `write` to save data itself, an arbitrary number of frames have executed, and the player may have
+loaded a new map, meaning we've written data to who knows where.
+
+There are two solutions to this problem.
+
+1. Use `guarded_write` instead of `write`. We can include a guard against the address changing, and the script will only
+perform the write if the data in memory matches what's in the guard. In the below example, `write_result` will be `True`
+if the guard validated and the data was written, and `False` if the guard failed to validate.
+
+```py
+# Get the address of the save data
+read_result: bytes = (await _bizhawk.read(ctx, [(0x301111, 4, "System Bus")]))[0]
+save_data_address = int.from_bytes(read_result, "little")
+
+# Write to `save_data_address` if it hasn't changed
+write_result: bool = await _bizhawk.guarded_write(
+    ctx,
+    [(save_data_address, [0xAA, 0xBB], "System Bus")],
+    [(0x301111, read_result, "System Bus")]
+)
+
+if write_result:
+    # The data at 0x301111 was still the same value as
+    # what was returned from the first `_bizhawk.read`,
+    # so the data was written.
+    ...
+else:
+    # The data at 0x301111 has changed since the
+    # first `_bizhawk.read`, so the data was not written.
+    ...
+```
+
+2. Use `lock` and `unlock` (discouraged if not necessary). When you call `lock`, you tell the emulator to stop advancing
+frames and just process requests until it receives an unlock request. This means you can lock, read the address, write
+the data, and then unlock on a single frame. **However**, this is _slow_. If you can't get in and get out quickly
+enough, players will notice a stutter in the emulation.
+
+```py
+# Pause emulation
+await _bizhawk.lock(ctx)
+
+# Get the address of the save data
+read_result: bytes = (await _bizhawk.read(ctx, [(0x301111, 4, "System Bus")]))[0]
+save_data_address = int.from_bytes(read_result, "little")
+
+# Write to `save_data_address`
+await _bizhawk.write(ctx, [(save_data_address, [0xAA, 0xBB], "System Bus")])
+
+# Resume emulation
+await _bizhawk.unlock(ctx)
+```
+
+You should always use `guarded_read` and `guarded_write` instead of locking the emulator if possible. It may be
+unreliable, but that's by design. Most of the time you should have no problem giving up and retrying. Data that is
+volatile but only changes occasionally is the perfect use case.
+
+If data is almost guaranteed to change between frames, locking may be the better solution. You can lower the time spent
+locked by using `send_requests` directly to include as many requests alongside the `LOCK` and `UNLOCK` requests as
```

Comment:
I'd assume this set shouldn't be capitalized, since the other instances aren't
```suggestion
locked by using `send_requests` directly to include as many requests alongside the `lock` and `unlock` requests as
```

Reply:
The Python functions are `lock` and `unlock`, but the request types are `LOCK` and `UNLOCK`. I mean to refer to the values being put in the data that the reader might be sending using `send_requests`.